src/PlatformInstaller/DotNetUpgrade/RuntimeUpgradeManager.cs

```diff
@@ -0,0 +1,132 @@
+        installer = new FileInfo(Path.Combine(assetFolder.FullName, dotNet452WebInstallerExe));
+    }
+    
+    public bool Is452orLaterInstalled()
+    {
+        using (var ndpKey = RegistryKey.OpenBaseKey(RegistryHive.LocalMachine, RegistryView.Registry32).OpenSubKey(@"SOFTWARE\Microsoft\NET Framework Setup\NDP\v4\Full\"))
+        {
+            if (ndpKey?.GetValueKind("Release") == RegistryValueKind.DWord)
+            {
+                var release = (int) ndpKey.GetValue("Release");
+                return release >= 379893;
+            }
+        }
+        return false;
+    }
+
+    public async Task<FileInfo> Download452WebInstaller()
+    {
+        eventAggregator.PublishOnUIThread(new DotNetDownloadStartedEvent());
+        using (var client = new WebClient())
+        {
+            client.Proxy.Credentials = credentialStore.Credentials;
+            client.DownloadProgressChanged += OnClientOnDownloadProgressChanged;
+            client.DownloadFileCompleted += OnClientOnDownloadFileCompleted;
+
+            var retries = 0;
+            const int maxretries = 5;
+
+            while (true)
+            {
+                LogTo.Information("Attempting to download '{0}' to '{1}'", dotNet452WebInstallerURL, installer.FullName);
+                try
+                {
+                    await client.DownloadFileTaskAsync(dotNet452WebInstallerURL, installer.FullName).ConfigureAwait(false);
+                    return installer;
+                }
+                catch
+                {
+                    retries++;
+                    PublishFailed();
+                    Thread.Sleep(500);
+                    if (retries <= maxretries)
+                    {
+                        continue;
+                    }
+                    PublishAborted();
+                }
+            }
+        }
+    }
+
+    public async void InstallDotNet452()
+    {
+
+        var exitCode = await processRunner.RunProcess(installer.FullName,
+            "/passive",
+            // ReSharper disable once PossibleNullReferenceException
+            installer.Directory.FullName,
+            s => { },
+            s => { }).ConfigureAwait(false);
+        
+        if (exitCode == 0)
+        {
+            eventAggregator.PublishOnUIThread(new DotNetInstallCompleteEvent());
+        }
+        else
+        {
+            eventAggregator.PublishOnUIThread(new DotNetInstallFailedEvent { ExitCode = exitCode });
+        }
+    }
+    
+    void PublishFailed()
+    {
+        eventAggregator.PublishOnUIThread(new DotNetDownloadFailedEvent
+        {
+            Text = "Download failed. Retrying..."
+        });
+    }
+
+    void PublishAborted()
+    {
+        eventAggregator.PublishOnUIThread(new DotNetDownloadAbortedEvent
+        {
+            IsError = true,
+            Text = "Download did not complete. Installation Aborted"
```

Comment:
isnt the text redundant if we have a specific event for it? or do we intend to re-use this?